urle27f76cced_iconmaster5326_HPPPlus_tgz-pJ8-com_iconmaster_hppplus_HPPPlusJ8/src/com/iconmaster/hppplus/parse/element/block/ElementIf.java

```diff
@@ -0,0 +1,161 @@
+
+package com.iconmaster.hppplus.parse.element.block;
+
+import com.iconmaster.hppplus.HPPPlus;
+import com.iconmaster.hppplus.Parseable;
+import com.iconmaster.hppplus.SourceRange;
+import com.iconmaster.hppplus.exception.HPPPlusException;
+import com.iconmaster.hppplus.exception.parse.EndExpectedException;
+import com.iconmaster.hppplus.exception.parse.SyntaxException;
+import com.iconmaster.hppplus.exception.parse.ThenExpectedException;
+import com.iconmaster.hppplus.parse.Element;
+import com.iconmaster.hppplus.parse.Parser;
+import com.iconmaster.hppplus.tokenize.Token;
+import com.iconmaster.hppplus.tokenize.element.TokenWord;
+import java.util.ArrayList;
+
+
+/**
+ *
+ * @author iconmaster
+ */
+public class ElementIf implements Element {
+    public class IfData {
+        private Element cond;
+        private ArrayList<Element> thenb;
+        private ArrayList<Element> elseb;
+        public IfData(Element cond,ArrayList<Element> thenb,ArrayList<Element> elseb) {
+            this.cond = cond;
+            this.thenb = thenb;
+            this.elseb = elseb;
+        }
+    }
+        
+    private SourceRange range;
+    private Element cond;
+    private ArrayList<Element> thenb;
+    private ArrayList<Element> elseb;
+    private int blockSize;
+    
+    
+    public ElementIf() {
+        
+    }
+    
+    public ElementIf(Element cond,ArrayList<Element> thenb,ArrayList<Element> elseb,int blockSize,SourceRange range) {
+        this.range = range;
+        this.cond = cond;
+        this.thenb = thenb;
+        this.elseb = elseb;
+        this.blockSize = blockSize;
+    }
+
+    @Override
+    public boolean isMatch(ArrayList<Parseable> parseList, int at) {
+        if (at+3>=parseList.size()) {return false;}
+        if (parseList.get(at) instanceof TokenWord) {
+            Token e = (Token) parseList.get(at);
+            return "IF".equalsIgnoreCase((String) e.getParsedContent());
+        }
+        return false;
+    }
+
+    @Override
+    public Element match(ArrayList<Parseable> parseList, int at) throws HPPPlusException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.